sql/log.cc

```diff
@@ -1812,6 +1812,16 @@ binlog_flush_cache(THD *thd, binlog_cache_mngr *cache_mngr,
     if (using_trx && thd->binlog_flush_pending_rows_event(TRUE, TRUE))
       DBUG_RETURN(1);
 
+#ifdef WITH_WSREP
+    /* Wsrep transaction was BF aborted but it must replay because certification
+       succeeded. The transaction must not be written into binlog yet, it will
+       be done during commit after the replay. */
+    if (WSREP(thd) && wsrep_must_replay(thd))
```

Comment:
I worried that this patch could potentially have unwanted consequences to regular transactions.  Can we move this up at least one level, to binlog_commit_flush_stmt_cache?

Reply:
Binlog cache can contain only something from non-transactional tables as transactional table changes can not be there yet. In wsrep we do not allow transaction contain changes to both transactional and non-transactional tables. So in replay case it can be only sequences.

Reply:
I think this place is better than binlog_commit_flush_stmt_cache